JavaScript/FastNoiseLite.js

```diff
@@ -0,0 +1,3372 @@
+		this.#CalculateFractalBounding();
+	}
+
+	/**
+	 * @description Sets octave lacunarity for all fractal noise types
+	 * @remarks Default: 2.0
+	 * @default 2.0
+	 * @param {number} lacunarity
+	 */
+	SetFractalLacunarity(lacunarity) {
+		this.#Lacunarity = lacunarity;
+	}
+
+	/**
+	 * @description Sets octave gain for all fractal noise types
+	 * @remarks Default: 0.5
+	 * @default 0.5
+	 * @param {number} gain
+	 */
+	SetFractalGain(gain) {
+		this.#Gain = gain;
+		this.#CalculateFractalBounding();
+	}
+
+	/**
+	 * @description Sets octave weighting for all none DomainWarp fratal types
+	 * @remarks Default: 0.0 | Keep between 0...1 to maintain -1...1 output bounding
+	 * @default 0.5
+	 * @param {number} weightedStrength
+	 */
+	SetFractalWeightedStrength(weightedStrength) {
+		this.#WeightedStrength = weightedStrength;
+	}
+
+	/**
+	 * @description Sets strength of the fractal ping pong effect
+	 * @remarks Default: 2.0
+	 * @default 2.0
+	 * @param {number} pingPongStrength
+	 */
+	SetFractalPingPongStrength(pingPongStrength) {
+		this.#PingPongStrength = pingPongStrength;
+	}
+
+	/**
+	 * @description Sets distance function used in cellular noise calculations
+	 * @remarks Default: EuclideanSq 
+	 * @default FastNoise.CellularDistanceFunction.EuclideanSq
+	 * @param {FastNoise.CellularDistanceFunction} cellularDistanceFunction
+	 */
+	SetCellularDistanceFunction(cellularDistanceFunction) {
+		this.#CellularDistanceFunction = cellularDistanceFunction;
+	}
+
+	/**
+	 * @description Sets return type from cellular noise calculations
+	 * @remarks Default: Distance
+	 * @default FastNoise.CellularReturnType.Distance
+	 * @param {FastNoise.CellularReturnType} cellularReturnType
+	 */
+	SetCellularReturnType(cellularReturnType) {
+		this.#CellularReturnType = cellularReturnType;
+	}
+
+	/**
+	 * @description Sets the maximum distance a cellular point can move from it's grid position
+	 * @remarks Default: 1.0
+	 * @default 1.0
+	 * @param {number} cellularJitter
+	 */
+	SetCellularJitter(cellularJitter) {
+		this.#CellularJitterModifier = cellularJitter;
+	}
+
+	/**
+	 * @description Sets the warp algorithm when using DomainWarp(...)
+	 * @remarks Default: OpenSimplex2
+	 * @default FastNoise.DomainWarpType.OpenSimplex2
+	 * @param {FastNoise.DomainWarpType} domainWarpType
+	 */
+	SetDomainWarpType(domainWarpType) {
+		this.#DomainWarpType = domainWarpType;
+		this.#UpdateWarpTransformType3D();
+	}
+
+	/**
+	 * @description Sets the maximum warp distance from original position when using DomainWarp(...)
+	 * @remarks Default: 1.0
+	 * @default 1.0
+	 * @param {number} domainWarpAmp
+	 */
+	SetDomainWarpAmp(domainWarpAmp) {
+		this.#DomainWarpAmp = domainWarpAmp;
+	}
+
+	/**
+	 * @description 2D/3D noise at given position using current settings
+	 * @param {number} x X coordinate
+	 * @param {number} y Y coordinate
+	 * @param {number} [z] Z coordinate
+	 * @return {number} Noise output bounded between -1...1
+	 */
+	GetNoise(x, y, z) {
```

Comment:
Another way to do this could be a switch statement on `arguments`, with cases for 2 and 3 that contain the respective code directly. I think this would be a better solution for public methods, because it doesn't incur the cost of a deeper function call. For the private methods, I would stick to separately-named definitions.

Reply:
Even if the performance is the same on modern VMs, these methods should definitely not be allocated within the function body.

Also, overhead on function call is super small and modern VMs will inline code if it's hot and makes sense w/caches, etc.

Reply:
See also: https://v8.dev/blog/adaptor-frame which implies that in general var-arg ("overloaded") functions in JS are slower than individual methods with static signatures. If performance is the goal then the methods should be separated anyway. A "routing" function like `GetNoise` is still nice for simplicity and DX, but access to the statically defined methods should be given to API consumers as well.